tahoe_gui/tahoe.py

```diff
@@ -15,48 +15,39 @@
     os.environ["PATH"] += os.pathsep + os.path.join(
         os.path.dirname(sys.executable), 'Tahoe-LAFS')
 
+class TahoeEnvironment:
+    """
+    I represent a virtualenv that has Tahoe-LAFS installed in it.
+    """
+    def __init__(self, basedir):
+        self._base = basedir
 
-class CommandProtocol(ProcessProtocol):
-    def __init__(self, parent, callback_trigger=None):
-        self.parent = parent
-        self.trigger = callback_trigger
-        self.done = Deferred()
-
-    def outReceived(self, data):
-        for line in data.decode('utf-8').strip().split('\n'):
-            self.parent.out_received(line)
-            if not self.done.called and self.trigger and self.trigger in line:
-                self.done.callback(None)
-
-    def errReceived(self, data):
-        for line in data.decode('utf-8').strip().split('\n'):
-            self.parent.err_received(line)
-
-    def processEnded(self, reason):
-        if not self.done.called:
-            self.done.callback(None)
-
-    def processExited(self, reason):
-        if not self.done.called and not isinstance(reason.value, ProcessDone):
-            self.done.errback(reason)
+    @inlineCallbacks
+    def create(self):
+        """
+        :return: a Deferred that fires when we've installed a virtualenv
+        """
 
 
-class Tahoe():
-    def __init__(self, nodedir=None):
-        self.nodedir = nodedir
+class TahoeClient:
+    def __init__(self, nodedir):
```

Comment:
Oh, you meant for `Tahoe` to represent the actual tahoe command? Hmm, maybe then it should be the thing that manages the virtualenv?

I was thinking of it like this: 

 - `TahoeClient` represents the actual "tahoe client" that the GUI is responsible for running -- so it would take a `TahoeEnvironment` and a `node_dir`. 
 - A `TahoeEnvironment` would be the thing that builds/encapsulates the virtualenv, and knows the exact path to the "tahoe" program.

(So you'd create a `TahoeEnvironment` as part of the "install" step, and then later give it to the `TahoeClient` when you reach the "run the daemon" step).